dali/test/python/test_external_source_cupy.py

```diff
@@ -32,3 +34,45 @@ def test_external_source_with_iter_cupy_stream():
 
             for i in range(10):
                 check_output(pipe.run(), [np.array([attempt * 100 + i * 10 + 1.5], dtype=np.float32)])
+import sys
+def discard_stderr():
+    """
+    Discards error output of a routine if invoked as:
+
+    with discard_stderr():
+        ...
+    """
+    with open(os.devnull, 'w') as bit_bucket:
+        try:
+            stderr_fileno = sys.stderr.fileno()
+            old_stderr = os.dup(stderr_fileno)
+            try:
+                os.dup2(bit_bucket.fileno(), stderr_fileno)
+                yield
+            finally:
+                os.dup2(old_stderr, stderr_fileno)
+        except AttributeError:
+            # On some systems is stderr not a file descriptor but actually a virtual pipeline
+            # that can not be copied
+            yield
+
+def test_external_source_mixed_continuous():
+    batch_size = 2
+    iterations = 4
+    def generator(i):
+        if i % 2:
+            return cp.array([100 + i * 10 + 1.5] * batch_size, dtype=cp.float32)
+        else:
+            return batch_size * [cp.array([100 + i * 10 + 1.5], dtype=cp.float32)]
+
+    pipe = Pipeline(batch_size, 3, 0)
+
+    pipe.set_outputs(fn.external_source(device="gpu", source=generator, no_copy=True))
+    pipe.build()
+
+    pattern = "ExternalSource operator should not mix continuous and noncontinuous inputs. " \
+              "In such case additional memory that gather provided data in continuous space " \
+              "will be trashed."
+    with check_output_pattern(pattern):
+        for _ in range(iterations):
```

Comment:
do you need the loop here? Shouldn't it fail at the first iteration?

Reply:
No. I(t fails when you provide contiguous first and then non-contiguous memory to the ExternalSource. So at lest two iterations are needed, when the first input is non-contiguous, then it is 3. Making it 4 just to be sure.